packages/mdast-util-split/__tests__/index.js

```diff
@@ -118,98 +118,11 @@ test('default parameter with canonical text', () => {
   })
 })
 
-test('extract conclusion', () => {
-  const result = doSplit(text, {conclusionAsProperty: true})
-  expect(result.introduction).toMatchObject({
-    type: 'root',
-    children: [
-      {
-        type: 'paragraph',
-        children: [
-          {
-            type: 'text',
-            value: 'a global introduction',
-          },
-        ],
-      },
-    ],
-  })
-  expect(result.trees.length).toBe(1)
-  expect(result.trees[0].children).toMatchObject({
-    type: 'root',
-    children: [
-      {
-        type: 'paragraph',
-        children: [
-          {
-            type: 'text',
-            value: 'a paragraph',
-          },
-        ],
-      },
-      {
-        type: 'blockquote',
-        children: [
-          {
-            type: 'paragraph',
-            children: [
-              {
-                type: 'text',
-                value: 'a quote to ',
-              },
-              {
-                type: 'emphasis',
-                children: [
-                  {
-                    type: 'text',
-                    value: 'ensure this is parsed',
-                  },
-                ],
-              },
-            ],
-          },
-        ],
-      },
-      {
-        type: 'heading',
-        depth: 2,
-        children: [
-          {
-            type: 'text',
-            value: 'a sub title',
-          },
-        ],
-      },
-      {
-        type: 'paragraph',
-        children: [
-          {
-            type: 'text',
-            value: 'other paragraph',
-          },
-        ],
-      },
-    ],
-  })
-  expect(result.conclusion).toMatchObject({
-    type: 'root',
-    children: [
-      {
-        type: 'paragraph',
-        children: [
-          {
-            type: 'text',
-            value: 'paragraph',
-          },
-        ],
-      },
-    ],
-  })
-})
-
 test('no heading', () => {
-  const noHeadingText = text.replace(/#/g, '')
-  expect(() => doSplit(noHeadingText, {})).toThrowError('No heading')
+  const headingStripped = text.replace(/#/g, '')
+  const result = doSplit(headingStripped, {})
+  expect(result.trees).toHaveLength(0)
+  expect(result.introduction.type).toBe('root')
```

Comment:
so you changed the behaviour?

Reply:
Yep, I forgot to mention this change. Throwing makes things more complicated for no real benefit.

I'm not sure about your use case for this but people should rightfully expect this package to split at the depth they have specified and get the appropriate result whether the specified depth exists or not. Some people might want to automate this split for exports for instance, the content before the first heading to split is considered the introduction. This should not fail.

Another way to look at it is:

```shell
$ python
>>> 'abc'.split(',')
['abc']
$ node
> 'abc'.split(',')
[ 'abc' ]
```

Reply:
true enouh. I merge your PR. Just need to update the doc now :)